src/genophenocorr/view/_protein_visualizer.py

```diff
@@ -146,29 +161,22 @@ def draw_fig(self, pvis:ProteinVisualizable):
         protein_track_y_min, protein_track_y_max = 0.492, 0.508
         font_size = 12
         text_padding = 0.004
-        
-        apprx_n_x_ticks = 6
-        min_x_absolute = minimum_aa_pos
-        max_x_absolute = maximum_aa_pos
-        tick_step_size, base = round_to_nearest_power_ten((max_x_absolute - min_x_absolute) / apprx_n_x_ticks)
-        print(base)
-        x_ticks = np.array(list(filter(
-            lambda x: x < max_x_absolute,
-            [min_x_absolute + i * tick_step_size for i in range(1, apprx_n_x_ticks + 1)]
-        ))).astype(int)
-        x_ticks = round_to_nearest_power_ten(x_ticks, base)
 
         max_marker_count = np.max(marker_counts)
 
+        x_ticks = generate_ticks(apprx_n_ticks=6, min=min_aa_pos, max=max_aa_pos)
+        y_ticks = generate_ticks(apprx_n_ticks=5, min=0, max=max_marker_count)
+
         # normalize into [0, 1], leaving some space on the sides
-        def preprocess(x_absolute):
-            print(f"type {type(x_absolute)} ")
-            print(f"type max_x_absolute {type(max_x_absolute)} ")
-            if x_absolute.any() > max_x_absolute:
-                print(f"x={x_absolute} but max={max_x_absolute}")
-            shifted_to_0_1 = ((x_absolute - min_x_absolute) / (max_x_absolute - min_x_absolute))
-            relative_scale = (protein_track_x_max - protein_track_x_min)
-            return shifted_to_0_1 * relative_scale + protein_track_x_min
+        def preprocess(absolute,
+                       min_absolute=min_aa_pos, max_absolute=max_aa_pos,
+                       min_relative=protein_track_x_min, max_relative=protein_track_x_max):
+            if absolute.any() > max_absolute:
```

Comment:
The context suggests that `absolute` is a numpy array with `float`s or `int`s. What is the meaning of calling `.any()` on such array?

Reply:
But then it should be:
```python
(absolute > max_absolute).any()
```

I'll fix it, thanks a lot for the info.

Reply:
With respect to the 5 or 3UTR variants, we should we put them at the start (`0`) or the end (`len(protein)`)  coordinates? Basically do the same thing we intended for intronic variants, putting them close to the place where they are relevant?

Please shout if this is a bad idea, otherwise I will place the variants there, so that we show *all* variants in the figure.